framework/src/main/java/org/checkerframework/framework/type/TypeFromExpressionVisitor.java

```diff
@@ -224,8 +225,17 @@ public AnnotatedTypeMirror visitArrayAccess(ArrayAccessTree node, AnnotatedTypeF
             f.visitorState.setAssignmentContext(null);
 
             AnnotatedTypeMirror type = f.getAnnotatedType(node.getExpression());
-            assert type instanceof AnnotatedArrayType;
-            return ((AnnotatedArrayType) type).getComponentType();
+            if (type instanceof AnnotatedArrayType) {
+                return ((AnnotatedArrayType) type).getComponentType();
+            } else if (type instanceof AnnotatedWildcardType) {
+                // Clean-up after Issue #979.
+                AnnotatedWildcardType wc = (AnnotatedWildcardType) type;
```

Comment:
Do you expect this to be an uninferred type argument? If so should we check that it is?